### PR TITLE
Add asc keyword to sort command

### PR DESCRIPTION
**Pull request by ritvibhatt**
### Description
- Adds asc/a keyword as option in sort command, Specifying asc does not change behavior, sort direction of each field remains as specified (ascending as default)
- Adds missing tests to ```CaclitePPlSortIT``` for desc keyword, count, and type casting enhancements

## Changes
  - Added `ASC` and `A` tokens to PPL lexer and parser grammar
  - Updated sort command to accept `asc`/`a` keywords alongside existing `desc`/`d`

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] New PPL command [checklist](https://github.com/opensearch-project/sql/blob/main/DEVELOPER_GUIDE.rst#new-ppl-command-checklist) all confirmed.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff` or `-s`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Swiddis — COMMENTED** [review]
Needs doc, otherwise LGTM

**Review by Swiddis — DISMISSED** [review]

File: docs/user/ppl/cmd/sort.rst [hunk 1]
```diff
@@ -16,13 +16,13 @@ Description
...
+* count (Since 3.3): optional. The number of results to return. **Default:** returns all results. Specifying a count of 0 or less than 0 also returns all results.
 * [+|-]: optional. The plus [+] stands for ascending order and NULL/MISSING first and a minus [-] stands for descending order and NULL/MISSING last. **Default:** ascending order and NULL/MISSING first.
 * sort-field: mandatory. The field used to sort. Can use ``auto(field)``, ``str(field)``, ``ip(field)``, or ``num(field)`` to specify how to interpret field values.
-* [desc|d]: optional. Reverses the sort results. If multiple fields are specified, reverses order of the first field then for all duplicate values of the first field, reverses the order of the values of the second field and so on.
+* [asc|a|desc|d] (Since 3.3): optional. asc/a keeps the sort order as specified. desc/d reverses the sort results. If multiple fields are specified with desc/d, reverses order of the first field then for all duplicate values of the first field, reverses the order of the values of the second field and so on. **Default:** asc.
```
Comment (RyanL1997) [inline docs/user/ppl/cmd/sort.rst:25]: nit: Consider rewording to 'sorts in ascending order' for clarity.

File: docs/user/ppl/cmd/sort.rst [hunk 2]
```diff
@@ -16,13 +16,13 @@ Description
 
 Syntax
 ============
-sort [count] <[+|-] sort-field>... [desc|d]
+sort [count] <[+|-] sort-field>... [asc|a|desc|d]
 
 
-* count: optional. The number of results to return. **Default:** returns all results. Specifying a count of 0 or less than 0 also returns all results.
+* count (Since 3.3): optional. The number of results to return. **Default:** returns all results. Specifying a count of 0 or less than 0 also returns all results.
```
Comment (RyanL1997) [inline docs/user/ppl/cmd/sort.rst:22]: Do we have these covered in example section?

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 [hunk 3]
```diff
@@ -158,7 +158,7 @@ dedupCommand
    ;
 
 sortCommand
-   : SORT (count = integerLiteral)? sortbyClause (DESC | D)?
+   : SORT (count = integerLiteral)? sortbyClause (ASC | A | DESC | D)?
```
Comment (RyanL1997) [inline ppl/src/main/antlr/OpenSearchPPLParser.g4:161]: Should we consider the precedence if someone accidentally specifies both asc and desc? The grammar allows it but behavior might be undefined.

File: integ-test/src/test/java/org/opensearch/sql/calcite/remote/CalcitePPLSortIT.java [hunk 4]
```diff
@@ -234,4 +234,100 @@ public void testSortDate() throws IOException {
         rows("Virginia", "2018-08-19 00:00:00"),
         rows("Dale", "2018-11-13 23:33:20"));
   }
+
+  @Test
+  public void testSortWithAscKeyword() throws IOException {
```
Comment (RyanL1997) [inline integ-test/src/test/java/org/opensearch/sql/calcite/remote/CalcitePPLSortIT.java:239]: nit: Consider adding negative test cases:
 - Invalid combinations like 'sort field asc desc'
 - Edge cases with count=0"

File: docs/user/ppl/cmd/sort.rst (hunk 1 above)
Comment (ritvibhatt) [reply docs/user/ppl/cmd/sort.rst:25]: The behavior for asc is it will keep whatever the specified field sort order was, so for ```sort firstname, -lastname asc``` firstname will still be sorted ascending and lastname will still be sorted descending. Don't know if 'sorts in ascending order' would make that behavior clear

File: docs/user/ppl/cmd/sort.rst (hunk 2 above)
Comment (ritvibhatt) [reply docs/user/ppl/cmd/sort.rst:22]: Yes! just added 'Since 3.3' in this PR but added examples in previous PR

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 (hunk 3 above)
Comment (ritvibhatt) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4:161]: Just checked using both asc and desc will result in an error

Comment (RyanL1997) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4:161]: got it. Thanks for confirming that.

**Review by penghuo — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]

**Review by Swiddis — APPROVED** [review]
